render/egl.c

```diff
@@ -712,23 +714,39 @@ static char *get_render_name(const char *name) {
 }
 
 int wlr_egl_dup_drm_fd(struct wlr_egl *egl) {
-	if (egl->device == EGL_NO_DEVICE_EXT || !egl->exts.EXT_device_drm) {
+	if (egl->device == EGL_NO_DEVICE_EXT || (!egl->exts.EXT_device_drm &&
+			!egl->exts.EXT_device_drm_render_node)) {
 		return -1;
 	}
 
-	const char *primary_name = egl->procs.eglQueryDeviceStringEXT(egl->device,
-		EGL_DRM_DEVICE_FILE_EXT);
-	if (primary_name == NULL) {
-		wlr_log(WLR_ERROR,
-			"eglQueryDeviceStringEXT(EGL_DRM_DEVICE_FILE_EXT) failed");
-		return -1;
+	char *render_name = NULL;
+#ifdef EGL_EXT_device_drm_render_node
```

Comment:
Why the preprocessor? Isn't checking `egl->exts.EXT_device_drm_render_node` enough?

Reply:
`EGL_EXT_device_drm_render_node` has been merged recently, so won't be widely available in the system headers for a while. Need to wait for new libglvnd and Mesa releases.

Reply:
So we're waiting for a release before merging?

Reply:
No need, the `#ifdef` ensures this code is disabled for older Mesa/libglvnd, and is enabled for recent ones.